tests/unit/generateCertificateNumber.unitTest.ts

```diff
@@ -52,4 +53,51 @@ describe("TestResultsService calling generateExpiryDate", () => {
         });
     });
 
+    context("when inserting a testResult that is a vehicleType of hgv and it contains at least one Roadworthiness test type and the test result on the Roadworthiness test type is fail", () => {
```

Comment:
I reverted the code changes to service, but kept these new tests and ran them locally. I expect the unit tests to fail in such a situation.  They prove that the code change has fixed the issue.  Something isn't right.

Reply:
The test was not setting the test type classification; that's why it was not changing the result.
I rewrote the tests and added more cases